state/accumulator_test.go

```diff
@@ -417,86 +416,6 @@ func TestAccumulatorDupeEvents(t *testing.T) {
 	}
 }
 
-// Regression test for corrupt state snapshots.
-// This seems to have happened in the wild, whereby the snapshot exhibited 2 things:
-//   - A message event having a event_replaces_nid. This should be impossible as messages are not state.
-//   - Duplicate events in the state snapshot.
-//
-// We can reproduce a message event having a event_replaces_nid by doing the following:
-//   - Create a room with initial state A,C
-//   - Accumulate events D, A, B(msg). This should be impossible because we already got A initially but whatever, roll with it, blame state resets or something.
-//   - This leads to A,B being processed and D ignored if you just take the newest results.
-//
-// This can then be tested by:
-//   - Query the current room snapshot. This will include B(msg) when it shouldn't.
-func TestAccumulatorMisorderedGraceful(t *testing.T) {
```

Comment:
Why are we getting rid of this---is it superseded by the new test?

Reply:
Because the way it was trying to repro the failure mode is impossible now, and we have other tests which repro corrupt state snapshots.